src/org/hitachivantara/ci/jenkins/MinionHandler.groovy

```diff
@@ -294,6 +300,19 @@ class MinionHandler {
     else if (buildProperties[BRANCH_NAME]) {
       buildProperties[LIB_CACHE_ROOT_PATH] = "${buildProperties[LIB_CACHE_ROOT_PATH]}/${buildProperties[BRANCH_NAME]}"
     }
+
+    // when in a multibranch minion build, check if it's a PR or a branch build to see which slack channel to send
+    // a notification if intended
+    if (buildProperties[IS_MULTIBRANCH_MINION]) {
+      def slackChannel = buildProperties[CHANGE_ID] ? buildProperties[PR_SLACK_CHANNEL] : buildProperties[SLACK_CHANNEL]
+      if (slackChannel) {
+        buildProperties[SLACK_INTEGRATION] = true
+        buildProperties[SLACK_CHANNEL] = slackChannel
+        buildProperties[BUILD_PLAN_ID] = buildProperties[BUILD_PLAN_ID] + '@' +
+          (buildProperties[CHANGE_ID] ? 'PR-' + buildProperties[CHANGE_ID] : buildProperties[BRANCH_NAME]) as String
```

Comment:
You can just use BRANCH_NAME here, it will be PR-123 when we're dealing with a pull request.

Reply:
changed